File: src/main/java/com/insogroup/utils/initialisator/exceptions/InitializableException.java
package com.insogroup.utils.initialisator.exceptions;

import java.io.IOException;

/**
 * Exception will called when object can't be initialised
 */
public class InitializableException extends IOException {
    public InitializableException() {
    }

    public InitializableException(String message) {
        super(message);
    }

    public InitializableException(String message, Throwable cause) {
        super(message, cause);
    }

    public InitializableException(Throwable cause) {
        super(cause);
    }
}

File: src/main/kotlin/com/github/insanusmokrassar/utils/initialisator/exceptions/InitializableException.kt
package com.github.insanusmokrassar.utils.initialisator.exceptions

import java.io.IOException

/**
 * Exception will called when object can't be initialised
 */
class InitializableException : IOException {
    constructor() {}

    constructor(message: String) : super(message) {}

    constructor(message: String, cause: Throwable) : super(message, cause) {}

    constructor(cause: Throwable) : super(cause) {}
}
